beps/0001-notifications-system/README.md

```diff
@@ -160,89 +168,202 @@ The following backend service interfaces are added as part of this proposal.
 ```ts
 // TODO - We may want to add an additional wrapping here with interfaces for Notification and NotificationParameters
 
-interface SendNotificationRequest {
-  receivers: SignalReceivers;
+export type NotificationRecipients =
+  | {
+      type: 'entity';
+      entityRef: string | string[]; // typically a user or group entity, otherwise `spec.owner` of the entity/entities
+    }
+  | {
+      type: 'broadcast'; // all users
+    };
 
-  /* TODO - please contribute :) */
-}
+export type NotificationType = 'undone' | 'done' | 'saved';
 
-interface SendNotificationResponse {
+export type Notification = {
   id: string;
-  receivers: SignalReceivers;
+  userRef: string;
+  title: string;
+  description: string;
+  origin: string;
+  link: string;
+  created: Date;
+  saved: boolean;
+  read?: Date;
```

Comment:
Why not just boolean?

Reply:
More possibilities to filter and also gather data on how people are reacting to the notifications

Reply:
I would keep it simple unless we have a scenario from the end-user perspective for it. 
This way we will never be able to collect all the data for a potential heuristic yet-to-come.

Reply:
Don't know how much simpler is setting a boolean value compared to setting a date or null?

Reply:
In general it's better to have an optional timestamp than a boolean for flags imo. It's basically the same complexity implementation wise but can be a lot more useful. I'd always suffix it with `at` though, so `readAt` here and `read_at` as column name if that applies. Probably worth doing the same for `saved` too.

Reply:
Why the `saved` is a boolean and not Date? If there is no use-case for it, it's overkill.